lib/ord.js

```diff
@@ -22,78 +23,143 @@ const defaults = require("./defaults");
+    const pendingServiceNames = [];
+
+    for (const definitionKey of Object.keys(csn.definitions)) {
+        const definitionObj = csn.definitions[definitionKey];
+        switch (definitionObj.kind) {
+            case CDS_ELEMENT_KIND.service: {
+                pendingServiceNames.push(definitionKey);
+                const apiResourceName = _handleApiResource(definitionKey, definitionObj);
+                if (apiResourceName) pendingApiResourceNames.push(apiResourceName);
                 break;
-            // TODO: should be rewritten
-            case CDS_ELEMENT_KIND.entity:
-                if (!key.includes(".texts")) {
-                    apiEndpoints.add(key);
-                    if (
-                        keyDefinition[ORD_ODM_ENTITY_NAME_ANNOTATION] ||
-                        keyDefinition[ENTITY_RELATIONSHIP_ANNOTATION]
-                    ) {
-                        entityTypeTargets.push(createEntityTypeMappingsItemTemplate(keyDefinition));
-                    }
+            }
+            case CDS_ELEMENT_KIND.entity: {
+                const result = _handleEntity(definitionKey, definitionObj);
+                if (result) {
+                    if (result.apiEndpoint) apiEndpoints.add(result.apiEndpoint);
+                    if (result.entityTypeTarget) entityTypeTargets.push(result.entityTypeTarget);
                 }
                 break;
-            case CDS_ELEMENT_KIND.event:
-                events.push(key);
+            }
+            case CDS_ELEMENT_KIND.event: {
+                const event = _handleEvent(definitionKey);
+                if (event) pendingEventNames.push(event);
                 break;
+            }
             case CDS_ELEMENT_KIND.action:
-            case CDS_ELEMENT_KIND.function:
-                apiEndpoints.add(key);
+            case CDS_ELEMENT_KIND.function: {
+                const apiEndpoint = _handleActionOrFunction(definitionKey);
+                if (apiEndpoint) apiEndpoints.add(apiEndpoint);
                 break;
+            }
         }
     }
 
     return {
-        env: cds.env["ord"],
-        lastUpdate,
-        appName,
+        serviceNames: pendingServiceNames,
+        apiResourceNames: pendingApiResourceNames,
         apiEndpoints: Array.from(apiEndpoints),
-        events,
-        serviceNames,
-        entityTypeTargets: _.uniqBy(entityTypeTargets, CONTENT_MERGE_KEY),
-        ordNamespace,
-        eventApplicationNamespace,
-        packageName,
+        eventNames: pendingEventNames,
+        entityTypeTargets,
     };
-};
+}
+
+function _handleApiResource(apiResourceName, serviceDefinition) {
+    if (_shouldSkipIfServiceOnlyContainsEvents(serviceDefinition) || serviceDefinition["@cds.external"]) {
+        return null;
+    }
+    return apiResourceName;
+}
 
-const _getPolicyLevels = (
-    appConfig) => appConfig.env?.policyLevels ||
-    appConfig.env?.policyLevel && [appConfig.env?.policyLevel] ||
-    defaults.policyLevels;
```

Comment:
is this intended? I had introduced the fallback handling here, if someone still uses policyLevel singular.

Reply:
well, there is a difference between the ORD compatibility and our plugins config compatible. therefore, removing it has the risk of disturbing a first adopter